vars/mavenCanaryRelease.groovy

```diff
@@ -7,6 +7,9 @@ import org.apache.maven.model.Profile
 
 @Field final String FMP_STABLE_VERSION = "3.5.38"
 
+// First version of FMP to include 'osio' profile
+@Field final String FMP_OSIO_MIN_VERSION = "3.5.40"
```

Comment:
@piyush1594 says "3.5.40" is stable now.

Reply:
Yes it is a stable version and also maintaining the two FMP versions here is not a good thing

Reply:
@jaseemabid @piyush1594 If the auto-update of FMP is disabled, or does not update for any reason, this would still allow us to apply space labels for FMP versions >= 3.5.40 but less than FMP_STABLE_VERSION.

FMP_OSIO_MIN_VERSION should never need to change. It should only used to check that the `osio` FMP profile exists. I could reduce the scope of this variable to the function it is used in. What do you think?

Reply:
@ebaron Got your point. you want this to check that profile extend feature is there or not that is available from FMP 3.5.40. Cool then.

Please also change the FMP Stable version to 3.5.40, changes in boosters are also in the process so.

Reply:
@piyush1594 Right, 3.5.40 brings the profile extends functionality, the `osio` profile, and the enricher for applying space labels. All of these things are required for this PR.

For better clarity, I have renamed the FMP_OSIO_MIN_VERSION constant to FMP_OSIO_PROFILE_MIN_VERSION, and added a comment specifying that it should not be changed.

I would gladly update the stable version to 3.5.40, but I encountered a slight problem. We also need to change the reference versions of booster-parent and spring-boot-booster-parent used by comparisons in `patchPluginBasedOnParent` and `patchProfileBasedOnParent`. The booster-parent update has been released as version 23, but there is no such release yet for spring-boot-booster-parent. I'm not sure if this will be released as 1.5.13-3 or 1.5.14. Any idea?

https://github.com/fabric8io/fabric8-pipeline-library/blob/205dcb720aea989c647b145262c461ef9f09b05c/vars/mavenCanaryRelease.groovy#L190-L194

https://github.com/fabric8io/fabric8-pipeline-library/blob/205dcb720aea989c647b145262c461ef9f09b05c/vars/mavenCanaryRelease.groovy#L212-L216